include/common/oclengine.hpp

```diff
@@ -140,14 +141,20 @@ class OCLDeviceContext {
         if (error != CL_SUCCESS) {
             queue = cl::CommandQueue(context, d);
         }
+
+        wait_events =
+            std::shared_ptr<std::vector<cl::Event>>(new std::vector<cl::Event>(), [](std::vector<cl::Event>* vec) {
```

Comment:
Consider make_shared instead?

Reply:
Also, not quite sure I understand why you need to explicitly specify this destructor functor.

Reply:
The `cl::Event` destructors _must_ be called. `clear()` will do this. I'm not sure if `delete` alone on the vector requires it.

Reply:
Right, and the other part is that only one copy of the same `cl::Event` owns the reference count, (even without shared pointers on top,) and the destructor has to be called once for the zeroing of the ref count. The issue seems to have been that `std::vector` on the ARM used a copying method that wasn't the one expected by the OpenCL headers, for transfering ownership of the reference count. The point is, this has no dependence on the operator used to copy elements of a `std::vector.` So, it works on all four test devices I own, whereas the Xeon and the Raspberry Pi were hard to find a common implementation for.